app/models/user.rb

```diff
@@ -85,6 +85,8 @@ def self.name_format
   validate :name_used_in_a_usergroup, :ensure_hidden_users_are_not_renamed, :ensure_hidden_users_remain_admin,
            :ensure_privileges_not_escalated, :default_organization_inclusion, :default_location_inclusion,
            :ensure_last_admin_remains_admin, :hidden_authsource_restricted, :ensure_admin_password_changed_by_admin
+  before_validation :verify_current_password, :if => Proc.new {|user| user == User.current},
+                    :unless => Proc.new {|user| user.password.empty?}
```

Comment:
@domcleal When user want to change some attribute without changing password. (e.g to change his own description or location without entering current password)